tools/toolchains/arm.py

```diff
@@ -554,7 +568,12 @@ def __init__(self, target, *args, **kwargs):
                     "ARM/ARMC6 compiler support is required for ARMC6 build"
                 )
 
-        if getattr(target, "default_toolchain", "ARMC6") == "uARM":
+        if  (getattr(target, "default_lib", "std") == "small" and getattr(target, "microlib_supported", False) == False 
+                and getattr(target, "default_toolchain", "ARM") != "uARM"):    
+                raise NotSupportedException(
+                    "MicroLib is not supported for this target")
```

Comment:
It is not Microlib that is not supported. It is the option default_lib that is not supported for the target.
```suggestion
                    "target.default_lib" = "small" is not supported for this target")
```